workers/unity/Packages/io.improbable.worker.sdk/CEventTrace.cs

```diff
@@ -0,0 +1,396 @@
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_Create")]
+        public static extern EventData EventDataCreate();
+
+        /** Frees resources for the event data object.*/
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_Destroy")]
+        public static extern void EventDataDestroy(IntPtr data);
+
+        /**
+         * Adds the key value pair as a field to the given event data object. Note that this may invalidate
+         * any keys or values retrieved with EventData_Get*.
+         */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_AddStringFields")]
+        public static extern void EventDataAddStringFields(EventData data, Uint32 count, Char** keys, Char** values);
+
+        /** Returns the number of fields on the given event data object. */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_GetFieldCount")]
+        public static extern Uint32 EventDataGetFieldCount(EventData data);
+
+        /**
+         * Returns all the key value pairs in the event data object. keys and values must have capacity for
+         * at least Trace_EventData_GetFieldCount(data) elements. This method is provided to discover key
+         * value pairs of unknown event schema data, therefore the ordering of key value pairs is entirely
+         * arbitrary.
+         */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_GetStringFields")]
+        public static extern void EventDataGetStringFields(EventData data, Char** keys, Char** values);
+
+        /** Returns the value for the given key. */
+        [DllImport(WorkerLibrary, CallingConvention = CallingConvention.Cdecl,
+            EntryPoint = "Trace_EventData_GetFieldValue")]
+        public static extern Char* EventDataGetFieldValue(EventData data, Char* key);
+
+        /** Data for an event added to the event-tracer. */
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Event
+        {
+            public SpanId Id;
+            public Uint64 UnixTimestampMillis;
+            public Char* Message;
+            public Char* Type;
+
+            // Use the Trace_EventData_* methods to read the data.
+            public EventData Data;
+        }
+
+        /** An item added to the event-tracer. */
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Item
+        {
+            /** The type of the item, defined using ItemType. */
+            public Uint8 ItemType;
+
+            [StructLayout(LayoutKind.Explicit)]
+            public struct Union
+            {
+                [FieldOffset(0)] public Span Span;
+
+                [FieldOffset(0)] public Event Event;
+            }
+        }
```

Comment:
`Item` needs a member of type `Item.Union`. The header file both declares the union and has a field called `item`: 

```c
typedef struct Trace_Item {
  /** The type of the item, defined using Trace_ItemType. */
  uint8_t item_type;
  union Trace_Item_Union {
    Trace_Span span;
    Trace_Event event;
  } item;
} Trace_Item;
```

Reply:
Added the new field to the Item struct.